src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -0,0 +1,169 @@
+              new Delivery(manager.getFactoryController().getNewInstance(
+                pos,
+                new TypeToken<ILocation>() {}), request.getRequester().getDeliveryLocation(), request.getRequest())));
+        }
+
+        return Lists.newArrayList();
+    }
+
+    @Nullable
+    @Override
+    public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        //Noop delivery has been completed
+    }
+
+    @Nullable
+    @Override
+    public IRequest getFollowupRequestForCompletion(
+                                                     @NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> completedRequest)
+    {
+        //No followup needed.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public IRequest onParentCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        //TODO Release the stack from the warehouse.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public void onResolvingOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        //TODO Release the stack from the warehouse.
+    }
+
+    @SuppressWarnings("squid:S2972")
+    /**
+     * We have this class the way it is for a reason.
+     */
+    private final class WarehouseChestDeliveryRequester implements IRequester
+    {
+        private final WarehouseRequestResolver warehouseRequestResolver;
+        private final IToken                   id;
+        private final ILocation                location;
+        private final IToken                   itemStackRequestToken;
+
+        private WarehouseChestDeliveryRequester(
+                                                 final WarehouseRequestResolver warehouseRequestResolver,
+                                                 final IToken id,
+                                                 final ILocation location,
+                                                 final IToken itemStackRequestToken)
+        {
+            this.warehouseRequestResolver = warehouseRequestResolver;
+            this.id = id;
+            this.location = location;
+            this.itemStackRequestToken = itemStackRequestToken;
+        }
+
+        @Override
+        public IToken getID()
+        {
+            return id;
+        }
+
+        @NotNull
+        @Override
+        public ILocation getLocation()
+        {
+            return location;
+        }
+
+        @NotNull
+        @Override
+        public void onRequestComplete(@NotNull final IToken token)
+        {
+            warehouseRequestResolver.onRequestComplete(itemStackRequestToken);
+        }
+    }
+
+    @NotNull
+    @Override
+    public void onRequestComplete(@NotNull final IToken token)
+    {
+        //TODO Release the DMan that did the job.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)